packages/teleport/src/Discover/LoginTrait/LoginTrait.test.tsx

```diff
@@ -0,0 +1,115 @@
+/**
+ * Copyright 2022 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import type { RenderResult } from '@testing-library/react';
+import { render, screen, act, fireEvent } from 'design/utils/testing';
+import type { User } from 'teleport/services/user';
+import { DiscoverContext } from '../discoverContext';
+import ContextProvider from '../discoverContextProvider';
+import LoginTrait from './LoginTrait';
+
+describe('login trait comp behavior', () => {
+  const ctx = new DiscoverContext();
+  const userSvc = ctx.userService;
+
+  let Component;
+
+  beforeEach(() => {
+    Component = (
+      <ContextProvider value={ctx}>
+        <LoginTrait
+          // TODO we don't need all of this
+          attempt={null}
+          joinToken={null}
+          createJoinToken={null}
+          agentMeta={null}
+          updateAgentMeta={null}
+          nextStep={null}
+          prevStep={null}
+        />
+      </ContextProvider>
+    );
+  });
+
+  test('add a new login', async () => {
+    jest.spyOn(userSvc, 'fetchUser').mockResolvedValue(mockUser);
+
+    let r: RenderResult;
+    await act(async () => {
```

Comment:
render is internally wrapped in act so we don't have to do this manually
https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#wrapping-things-in-act-unnecessarily

Reply:
i found that you need them with async calls